.github/workflows/upstream-build.yml

```diff
@@ -0,0 +1,48 @@
+name: Silabs upstream daily build
+
+on:
+  schedule:
+    - cron: "15 5 * * *"
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, macos-14, windows-2022]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: zephyr-silabs
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install Silabs-specific python dependencies
+        run: |
+          pip3 install crc
+
+      - name: Setup Zephyr project
+        uses: zephyrproject-rtos/action-zephyr-setup@v1
+        with:
+          app-path: zephyr-silabs
+        toolchains: arm-zephyr-eabi
+
+      - name: Update manifest to point at upstream main branch
+        working-directory: zephyr-silabs
+        shell: bash
+        run: |
+          Z_REV=`grep -A 2 'name: zephyr$' west.yml|grep revision:|sed 's/^.*revision: //'`
+          sed -i .orig s/$Z_REV/main/ west.yml
+          west update
+          west blobs fetch hal_silabs
+
+      - name: Build Bluetooth sample for xG27
+        working-directory: zephyr-silabs
+        shell: bash
+        run: |
+          west build -b xg27_dk2602a/efr32bg27c140f768im40 ../zephyr/samples/bluetooth/peripheral_hr
```

Comment:
There is no way to use `twister` rather than relying on hardcoded values?

Reply:
> There is no way to use `twister` rather than relying on hardcoded values?

Part of the challenge is that almost all Bluetooth samples and tests specify an explicit `platform-allow` list, which they really shouldn't have, since there's an existing `bluetooth` tag that can be matched against instead. I'm hoping to improve the situation for Zephyr 4.0, and twister does have a "force" switch to force a specific platform, but I reverted back to basic `west build` for now since I wasn't able to figure out the appropriate twister command line. I guess you could take this as a rough starting point from where we can iterate further in subsequent pull requests.

Btw, I still didn't figure out how to manually trigger these "schedule" workflows. I may need to push a temporary version with a different trigger, and then once I see it runs correctly go back to the final "schedule" trigger.

Reply:
Seems like this is not something that will be fixed quickly, if you follow the discussion thread I started on #testing in Discord. Meanwhile, I think we should just use `west build`.